File: 3753-maximum-difference-between-even-and-odd-frequency-i/maximum-difference-between-even-and-odd-frequency-i.cpp
class Solution {
public:
    int maxDifference(string s) {
        vector<int> freq(26, 0);
        for (char c : s)
            freq[c - 'a']++;
        int f1 = 0, f2 = s.size();
        for (int val : freq) {
            if (val == 0)
                continue;
            if (val % 2 == 1)
                f1 = max(f1, val);
            else
                f2 = min(f2, val);
        }
        if (f2 == s.size())
            f2 = 0;
        return f1 - f2;
    }
};

File: 3753-maximum-difference-between-even-and-odd-frequency-i/maximum-difference-between-even-and-odd-frequency-i.java
class Solution {
    public int maxDifference(String s) {
        HashMap<Character, Integer> mp = new HashMap<>();
        for (char c : s.toCharArray()) {
            mp.put(c, mp.getOrDefault(c, 0) + 1);
        }

        int odd = 0, even = Integer.MAX_VALUE;
        for (char it : mp.keySet()) {
            int freq = mp.get(it);
            if (freq % 2 != 0)
                odd = Math.max(odd, freq);
            else
                even = Math.min(even, freq);
        }
        return odd - even;
    }
}